docs/usage/constrained.md

```diff
@@ -0,0 +1,151 @@
+
```

Comment:
again these should be moved next to the types they relate to, rather than but in a bucket on their own.